source/helpers.js

```diff
@@ -0,0 +1,47 @@
+const defineTypeWithFields = (definitons, BaseType, namespace, className, fields) => {
+  let classPath = className;
+  if (namespace !== '') classPath = `${namespace}.${className}`;
+  let SubType = BaseType.extend(classPath);
+  SubType.prototype.fields = function() {
+    return _.extend(BaseType.prototype.fields(), fields);
+  };
+  Space.resolvePath(namespace)[className] = SubType;
+  definitons[classPath] = SubType;
```

Comment:
Ah right. Seems a little odd to have that broken out into a separate function if it's just decorating a property, particularly since it's just 8 lines of code.